### ISSUE TITLE
Some questions on the frame when desync happens in online game

### ISSUE THREAD

**Issue by dnqbob** [OpenRA/OpenRA#21638]
1. Will the very desync frame recorded into replay?
2. if Q1 is satisfied, could we output SharedRandom, or even a entire sync-report on that frame after replay is done?
3. if Q1 & Q2 are satisfied, could we just use the replay, replay multiple time and generate many sync-reports to find out the desync place in code? (so the replay becomes a reproducing steps as it only issue order to OM)

**Comment by anvilvapre** [comment]
1. Not sure. Look at the code. It should be in the syncreport. In one of the log files.

2. 
`World.cs` shows:

`SharedRandom = new MersenneTwister(orderManager.LobbyInfo.GlobalSettings.RandomSeed);`

Based on this you can already reproduce the shared random. 

The global settings are in the replay file. 

```
GlobalSettings:                                                                                                                 ServerName: 64H Multiplayer - RA - VIII                                                                                 Map: 1da717f2a6cbabc6f7a93acec9e3b6aff94477aa                                                                           Timestep: 35                                                                                                    
        OrderLatency: 4                                                                                                 
        RandomSeed: -449298392                                                                                          
        AllowSpectators: True                                                                                           
        AllowVersionMismatch: False                                                                                     
        GameUid: 4cd788e5-97ee-45be-8793-e0a863998f93                                                                   
        EnableSingleplayer: True                                                                                        
        Dedicated: True                                                                                                 
        Options:                 

```

3. Yes, should be possible. Even on fastest speed it will take time.

4. Also dumping the network/replay frames and Orders in them can help debug.

**Comment by dnqbob** [comment]
Because current multiplayer debug base on the syncreport by players and they must toggle on "Generate Sync-report" option, which is too hard to get. 

If we can just use their replay to generate syncreport and Q1 is satisfied, then some of the desync problem (only related to incorrect c# coding) can be found out without asking players and depending on good luck.

**Comment by dnqbob** [comment]
So the key is, "Will the very desync frame recorded into replay?"

**Comment by anvilvapre** [comment]
No, I doubt it, not sure. Because the desync is where one client has a different state than the other.
So you would have the orders that your client received in the replay file. But you would still be missing their sync report of their state.

The number will be there.

The frame number of your client will be in the replay file - if the game did not exit before - but you will still miss their replay recording.
